src/main/java/frc/robot/commands/rests/restUtils/RESTHandler.java

```diff
@@ -0,0 +1,272 @@
+        timer = new Timer();
+    }
+
+    public void initialize() {
+        DataLogManager.start();
+        log = DataLogManager.getLog();
+
+        setupREST();
+
+        SendableRegistry.addLW(this, "RESTHandler");
+        timer.start();
+    }
+
+    public void initTest() {
+        init.run();
+    }
+
+    public void runTest() {
+        execute.run();
+    }
+
+    public void handleResults() {
+        String result;
+        try {
+            end.run();
+            result = String.format("%s :: PASSED\n", currentTest.getName().toUpperCase());
+            results[restIndex][testIndex] = result.stripTrailing();
+            resultLog.append(result);
+        } catch (Exception e) {
+            result = String.format("%s :: FAILED\n", currentTest.getName().toUpperCase());
+            results[restIndex][testIndex] = result.stripTrailing();
+            resultLog.append(result + "\t\t" + e.getLocalizedMessage());
+        }
+    }
+
+    public String[] getRESTResults(int restIndex) {
+        return results[restIndex];
+    }
+
+    public void reset() {
+        init = () -> {
+        };
+        execute = () -> {
+        };
+        end = () -> {
+        };
+
+        testFinished = true;
+        newTest = true;
+        timer.reset();
+    }
+
+    public void updateRESTProgress() {
+        if (testIndex < currentTestList.size() - 1) {
+            testIndex++;
+            currentTest = currentTestList.get(testIndex);
+            newREST = false;
+            return;
+        }
+
+        if (restIndex < rests.size() - 1) {
+            shutdownREST();
+            restIndex++;
+            testIndex = 0;
+            currentREST = rests.get(restIndex);
+            currentTestList = restTests.get(currentREST);
+            currentTest = currentTestList.get(testIndex);
+            setupREST();
+            newREST = true;
+        }
+    }
+
+    private void setupREST() {
+        for (Method method : currentREST.getClass().getDeclaredMethods()) {
+            if (method.isAnnotationPresent(Before.class)) {
+                method.setAccessible(true);
+                invokeMethod(method, currentREST);
+                break;
+            }
+        }
+        resultLog = new StringLogEntry(log, "RESTResult/" + currentREST.getClass().getSimpleName());
+    }
+
+    private void shutdownREST() {
+        for (Method method : currentREST.getClass().getDeclaredMethods()) {
+            if (method.isAnnotationPresent(After.class)) {
+                method.setAccessible(true);
+                invokeMethod(method, currentREST);
+                return;
+            }
+        }
+    }
+
+    public boolean isNewREST() {
+        return newREST;
+    }
+
+    public void updateRequirements() {
+        currentRequirements = new ArrayList<>();
+
+        Subsystem subsystem;
+        for (Field field : currentREST.getClass().getDeclaredFields()) {
+            if (field.isAnnotationPresent(Requirement.class)) {
```

Comment:
For situations like this where there is logic in the if statement, reversing the conditional and putting a continue statement in would reduce the indentation level and make it a little bit easier to read.

```
if (!field.isAnnotationPresent(Requirement.class)) {
	continue;
}
field.setAccessible(true);
try {
	subsystem = (Subsystem) field.get(currentREST);
	currentRequirements.add(subsystem);
} catch (IllegalAccessException e) {
	throw new RuntimeException(e);
}
```